docs/content/tutorial/step_05.ngdoc

```diff
@@ -1,277 +1,168 @@
+    </div>
+  </div>
 ```
 
-`$http` makes an HTTP GET request to our web server, asking for `phones/phones.json` (the url is
-relative to our `index.html` file). The server responds by providing the data in the json file.
-(The response might just as well have been dynamically generated by a backend server. To the
-browser and our app, they both look the same. For the sake of simplicity, we used a json file in this
-tutorial.)
-
-The `$http` service returns a {@link ng.$q promise object} with a `success`
-method. We call this method to handle the asynchronous response and assign the phone data to the
-scope controlled by this controller, as a model called `phones`. Notice that Angular detected the
-json response and parsed it for us!
-
-To use a service in Angular, you simply declare the names of the dependencies you need as arguments
-to the controller's constructor function, as follows:
-
-    phonecatApp.controller('PhoneListCtrl', function ($scope, $http) {...}
-
-Angular's dependency injector provides services to your controller when the controller is being
-constructed. The dependency injector also takes care of creating any transitive dependencies the
-service may have (services often depend upon other services).
-
-Note that the names of arguments are significant, because the injector uses these to look up the
-dependencies.
-
+We added a standard HTML `<input>` tag and used Angular's
+{@link ng.filter:filter filter} function to process the input for the
+{@link ng.directive:ngRepeat ngRepeat} directive.
 
-<img  class="diagram" src="img/tutorial/tutorial_05.png">
+By virtue of the {@link ng.directive:ngModel ngModel} directive, this lets a user enter search
+criteria and immediately see the effects of their search on the phone list. This new code
+demonstrates the following:
 
+* Data-binding: This is one of the core features in Angular. When the page loads, Angular binds the
+  value of the input box to the data model variable specified with `ngModel` and keeps the two in
+  sync.
 
-### `$` Prefix Naming Convention
+  In this code, the data that a user types into the input box (bound to **`$ctrl.query`**) is
+  immediately available as a filter input in the list repeater
+  (`phone in $ctrl.phones | filter:`**`$ctrl.query`**). When changes to the data model cause the
+  repeater's input to change, the repeater efficiently updates the DOM to reflect the current state
+  of the model.
 
-You can create your own services, and in fact we will do exactly that in step 11. As a naming
-convention, Angular's built-in services, Scope methods and a few other Angular APIs have a `$`
-prefix in front of the name.
+  <img  class="diagram" src="img/tutorial/tutorial_05.png">
 
-The `$` prefix is there to namespace Angular-provided services.
-To prevent collisions it's best to avoid naming your services and models anything that begins with a `$`.
+* Use of the `filter` filter: The {@link ng.filter:filter filter} function uses the `$ctrl.query`
```

Comment:
In this case, we need the `ng.filter:` part, because there is also `guide/filter`, so dgeni doesn't know what to pick for `filter`. (And as mentioned above, I chose not to format links as code for "distinguishability".)